### DISCUSSION TITLE
How would I pass data from the quantity input field to the event listener?

### DISCUSSION THREAD

**Discussion by aaronwjm360** [livewire/livewire#4110] [1. Help] [answered]
Here is an extract of the code in question:

```

<div>
   <label class="sr-only">Enter quantity</label>
   <input class="rounded-md placeholder-gray-500 text-center border-pink-300 mt-4 focus:ring-pink-400 focus:border-pink-400" type="number" id="quantity" name="quantity" min="1" max="100" value="1">
</div>
<button id="submit{{ $Product->id }}" wire:click="$emitTo('cart', 'add', {{ $Product }}, 'quantity_here')" x-on:click="cartSlideout = true" disabled class="cursor-default opacity-50 mt-4 block w-full bg-gradient-to-r from-slpink-dark via-slpink-light to-slpink-lighter hover:from-slpink-lighter hover:via-slpink-light hover:to-slpink-dark border border-transparent rounded-md py-2 text-md font-semibold text-white text-center">
ADD TO BAG
</button>
```

When I click the button, it successfully sends the model $Product over to the event listener. But I have an input field (quantity) which I need to pass to the listener also. How would I do this?

Thanks all!

**Comment by joshhanley** [comment]
@aaronwjm360 please format your code so it is easy to read, to make it easier to help you.

Thanks!

**Reply by aaronwjm360** [reply]
@joshhanley sorry about that, the formatting got screwed up when I copied and pasted. I've edited my original post, hopefully that helps! Thanks

**Reply by joshhanley** [reply]
No worries! Yep that makes it easier 🙂

**Comment by joshhanley** [comment]
Is this in a different component to the cart? 

So assuming you are looping over products, I would probably bind the quantity to an array like:

In your component class have 

```php
public $productQuantities = [];
```

Then bind to that in your blade code

```blade
<input type="text" wire:model="productsQuantities.{{ $product->id }}.quantity" />
```

Then change your `$emitTo` to actually call a method `wire:model="addToCart({{ $product->id }})"` and in that php method you would get the quantity from the array based on the product, then call `$emitTo`.

Hope this helps!

**Reply by aaronwjm360** [reply]
> Is this in a different component to the cart?
> 
> So assuming you are looping over products, I would probably bind the quantity to an array like:
> 
> In your component class have
> 
> ```
> public $productQuantities = [];
> ```
> 
> Then bind to that in your blade code
> 
> ```
> <input type="text" wire:model="productsQuantities.{{ $product->id }}.quantity" />
> ```
> 
> Then change your `$emitTo` to actually call a method `wire:model="addToCart({{ $product->id }})"` and in that php method you would get the quantity from the array based on the product, then call `$emitTo`.
> 
> Hope this helps!

Hey @joshhanley,

Sorry, i have another question. When using this example above, i would like the input field to default to the value of "1", so that the default quantity when adding the product to cart would be 1. Using your example above, how would I achieve this?

Kind Regards,

Aaron

**Comment by aaronwjm360** [comment]
Hi @joshhanley,

This sounds like it would work if the addToCart function and the product listing was in the same component, but they are actually 2 separate components (the cart is a slideover modal, activated using AlpineJS variables), so are you suggesting perhaps that I should just combine the addToCart function within the same Livewire component, leaving the Cart component to just deal with any functions on the cart page specifically?

I need these actions to then refresh the cart view/component, I guess I could just use emitTo in the Product component, within the PHP class itself?

Thanks so much for your help so far :)

Kind Regards,

Aaron

**Comment by joshhanley** [comment, marked as answer]
@aaronwjm360 in the same component would be simpler and would be my preference, that way you don't need to fire events around.

But two seperate components will work too.

It depends on how your product component is setup, do you have multiple product components on the page (one for each product) or do you just have a single product component that lists all the products?

If it's a product component per product, then you could simply add a quantity property to it and `wire:model` your input directly to that (rather than using the nested binding like I had above).

Then you would have a method on your product component that emits and event with the quantity.

Like this:

Product blade view
```blade
<input type="text" wire:model="quantity" />
<button type="button" wire:click="addToCart">Add to cart</button>
```

Product class
```php
public $product;
public $quantity;

public function addToCart()
{
    // You could even do some validation here too, that ensures that
    // there is enough of the product available before adding to cart

    $this->emitTo('cart', 'add', $this->product->id, $this->quantity);
}
```

Then in your Cart component, just listen for that `add` event and handle it as necessary.

Hope this helps!

**Comment by aaronwjm360** [comment]
Hi @joshhanley,

Thanks for your time and help on this - using your examples it seems to be working the way I need it to.

Livewire is amazing, keep up the good work man!

Kind Regards,

Aaron

**Reply by joshhanley** [reply]
No worries! Great to hear! 🙂
